python/GafferArnoldTest/IECoreArnoldPreviewTest/RendererTest.py

```diff
@@ -2310,6 +2311,44 @@ def testStatsAndLog( self ) :
 			driverEvents = [ x for x in stats if x["name"] == "ieCoreArnold:display:testBeauty" ]
 			self.assertEqual( driverEvents[0]["cat"], "driver_exr" )
 
+	def testMessageHandler( self ) :
+
+		RenderType = GafferScene.Private.IECoreScenePreview.Renderer.RenderType
+
+		for renderType, fileName, output in (
+			( RenderType.Batch, "", IECoreScene.Output( self.temporaryDirectory() + "/beauty.exr", "exr", "rgba" ) ),
+			( RenderType.Interactive, "", None ),
+			( RenderType.SceneDescription, self.temporaryDirectory() + "/test.ass" , None )
+		) :
+
+			with IECore.CapturingMessageHandler() as fallbackHandler :
+
+				handler = IECore.CapturingMessageHandler()
+
+				r = GafferScene.Private.IECoreScenePreview.Renderer.create(
+					"Arnold",
+					renderType,
+					fileName = fileName,
+					messageHandler = handler
+				)
+
+				r.option( "ai:console:info", IECore.BoolData( True ) )
+				r.option( "ai:console:progress", IECore.BoolData( True ) )
+				r.option( "ai:invalid", IECore.BoolData( True ) )
+
+				if output :
+					r.output( "testOutput", output )
+
+				r.render()
+				time.sleep( 1 )
+				del(r)
```

Comment:
Same comments as for AppleseedRender.

Reply:
This one is required in the end, as we need to ensure the previous renderer is deleted before the next iteration tries to construct one.